src/zenml/client.py

```diff
@@ -5062,50 +5157,179 @@ def get_model_version(
+
+        if len(model_versions) == 1:
+            return model_versions[0]
+        elif len(model_versions) == 0:
+            raise KeyError(
+                f"No model version found for model "
+                f"`{model_name_or_id}` with version identifier "
+                f"`{model_version_name_or_number_or_id}`."
+            )
+        else:
+            raise RuntimeError(
+                f"The model version identifier "
+                f"`{model_version_name_or_number_or_id}` is not"
+                f"unique for model `{model_name_or_id}`."
+            )
+
+    def list_model_versions(
+        self,
+        model_name_or_id: Optional[Union[str, UUID]] = None,
+        sort_by: str = "number",
+        page: int = PAGINATION_STARTING_PAGE,
+        size: int = PAGE_SIZE_DEFAULT,
+        logical_operator: LogicalOperators = LogicalOperators.AND,
+        created: Optional[Union[datetime, str]] = None,
+        updated: Optional[Union[datetime, str]] = None,
+        name: Optional[str] = None,
+        number: Optional[int] = None,
+        stage: Optional[Union[str, ModelStages]] = None,
+    ) -> Page["ModelVersionResponseModel"]:
         """Get model versions by filter from Model Control Plane.
 
         Args:
             model_name_or_id: name or id of the model containing the model version.
-            model_version_filter_model: All filter parameters including pagination
-                params.
+            sort_by: The column to sort by
+            page: The page of items
+            size: The maximum size of all pages
+            logical_operator: Which logical operator to use [and, or]
+            created: Use to filter by time of creation
+            updated: Use the last updated date for filtering
+            name: name or id of the model version.
+            number: number of the model version.
+            stage: stage of the model version.
 
         Returns:
-            A page of all model versions.
+            A page object with all model versions.
         """
+        model_version_filter_model = ModelVersionFilterModel(
+            page=page,
+            size=size,
+            sort_by=sort_by,
+            logical_operator=logical_operator,
+            created=created,
+            updated=updated,
+            name=name,
+            number=number,
+            stage=stage,
+        )
+
         return self.zen_store.list_model_versions(
             model_name_or_id=model_name_or_id,
             model_version_filter_model=model_version_filter_model,
         )
 
     def update_model_version(
         self,
-        model_version_id: UUID,
-        model_version_update_model: ModelVersionUpdateModel,
-    ) -> ModelVersionResponseModel:
+        model_name_or_id: Union[str, UUID],
+        version_name_or_id: Union[str, UUID],
+        stage: Optional[Union[str, ModelStages]] = None,
+        force: bool = False,
+        name: Optional[str] = None,
+    ) -> "ModelVersion":
```

Comment:
Still not a fan of this returning `ModelVersion` instead of `ModelVersionResponseModel`

Reply:
Let's proceed like this then:
- I will merge this one to unblock others (including you with rebasing your mega-branch).
- Spin a feedback loop on Discord to clarify what we should return and if we should stick to responses here.
- If we decide to use responses - I'll raise a separate ticket for that purpose.